### PR TITLE
CMR-7072 adding in GET DATA VIA DIRECT ACCESS as a new related URL type.

### PR DESCRIPTION
**Pull request by eereiter**
Updated ECHO10 <-> umm-g, UMM-G <-> UMM-G, UMM-G <-> umm-g, ISO SMAP -> umm-g, umm-g -> ISO MENDS

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj [hunk 1]
```diff
@@ -590,3 +590,78 @@
...
+
+    (util/are3 [granule number]
+      (let [gran-atom (da/granules->expected-atom [granule]
+                                                  [coll1]
+                                                  (str "granules.atom?granule_ur=Granule" number))
+            response (search/find-concepts-atom :granule {:granule-ur (str "Granule" number)})
+            links (get-in response [:results :links])]
+        (is (= 200 (:status response)))
+        (is (= (:links gran-atom) links))
+        (is (for [link links
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj:636]: This check is wrong. Use `some` function here.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj [hunk 2]
```diff
@@ -590,3 +590,78 @@
...
+                  :when (= "http://esipfed.org/ns/fedsearch/1.1/cloud" (:rel link))]
+              true)))
+
+      "atom link and the translation from old cmr umm-g to echo10 and back to old cmr umm-g."
+      gran-echo
+      1
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj:642]: just put `granule-ur` here to make the code more readable.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj [hunk 3]
```diff
@@ -590,3 +590,78 @@
...
+      3)
+
+    (testing "the translation from old cmr umm-g to UMM-G and back to old cmr umm-g without related URLs."
+      (let [gran-atom (da/granules->expected-atom [gran-umm] [coll1] "granules.atom?granule_ur=Granule4")
+            response (search/find-concepts-atom :granule {:granule-ur "Granule4"})
+            links (get-in response [:results :links])]
+        (def links links)
+        (is (= 200 (:status response)))
+        (is (= (:links gran-atom) links))
+        (is (= nil (seq
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj:659]: rewrite this check with `some`.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj [hunk 4]
```diff
@@ -590,3 +590,78 @@
...
+                                                  (str "granules.atom?granule_ur=Granule" number))
+            response (search/find-concepts-atom :granule {:granule-ur (str "Granule" number)})
+            links (get-in response [:results :links])]
+        (is (= 200 (:status response)))
+        (is (= (:links gran-atom) links))
+        (is (for [link links
+                  :when (= "http://esipfed.org/ns/fedsearch/1.1/cloud" (:rel link))]
+              true)))
+
+      "atom link and the translation from old cmr umm-g to echo10 and back to old cmr umm-g."
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj:640]: I find these comments confusing. This is just testing searching ECHO10 granule in ATOM format. Why do you mention `old umm-g`?

File: umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj [hunk 5]
```diff
@@ -120,6 +120,14 @@
   [list-of-maps]
   (seq (map #(downgrade-format-and-mimetype-to-1-6 %) list-of-maps)))
 
+(defn v1-6-2-related-url-type->1-6-1
+  "Migrate v1.6.2 related url Type to v1.6.1 related url Type"
+  [related-url]
+  (def related-url related-url)
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj:126]: Remove this line.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj [hunk 6]
```diff
@@ -181,3 +189,18 @@
...
+
+(defmethod interface/migrate-umm-version [:granule "1.6.2" "1.6.1"]
+  [context g & _]
+  (-> g
+      (assoc :MetadataSpecification {:URL "https://cdn.earthdata.nasa.gov/umm/granule/v1.6.1"
+                                     :Name "UMM-G"
+                                     :Version "1.6.1"})
+      (util/update-in-each [:RelatedUrls] v1-6-2-related-url-type->1-6-1)))
+
+(defmethod interface/migrate-umm-version [:granule "1.6.1" "1.6.2"]
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj:201]: Do you need to migrate existing s3 RelatedURLs that are `GET DATA` type in v1.6.1 to the new type in v1.6.2?

**Review by ygliuvt — CHANGES_REQUESTED** [review]
Looks good in general. Need some minor cleanup.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj (hunk 1 above)
Comment (eereiter) [reply system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj]: I have updated this function to use some.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj (hunk 2 above)
Comment (eereiter) [reply system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj]: I have updated the tests to use the granule-ur string and not just a number.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj (hunk 3 above)
Comment (eereiter) [reply system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj]: I have re-written this function to use some.

File: system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj (hunk 4 above)
Comment (eereiter) [reply system-int-test/test/cmr/system_int_test/search/granule_search_format_test.clj]: I have changed the comments to state that I am testing echo 10, or ISO smap, etc.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj (hunk 5 above)
Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj]: I removed the debugging def.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj (hunk 6 above)
Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/granule.clj:200]: No according to Doug Newman

**Review by ygliuvt — APPROVED** [review]

**Review by mschmele — APPROVED** [review]

**Pull request merged**